### PR TITLE
Install latest version of setuptools

### PR DESCRIPTION
**Pull request by stsewd**
Closes https://github.com/readthedocs/readthedocs.org/issues/7241

We have experienced other issues in the past bc of using an outdated version of setuptools too

### CODE REVIEW

**Review by humitos — DISMISSED** [review]
I think this is :+1: . We may want to roll this out gradually, tho. This change will affect _all_ the builds. Maybe adding a feature flag first, enable in some ~k projects and then remove the flag?

**Comment by stsewd** [comment]
I don't know, feels safe to upgrade that package. @readthedocs/core thoughts on using a feature flag for a while?

**Comment by ericholscher** [comment]
I think this is another "we need to figure out what to do with default versions" issue.

**Comment by stsewd** [comment]
This is under a feature flag now

File: readthedocs/rtd_tests/tests/test_doc_building.py [hunk 1]
```diff
@@ -1192,7 +1191,6 @@ def setUp(self):
         self.build_env_mock = Mock()
 
         self.base_requirements = [
-            'setuptools',
```
Comment (humitos) [inline readthedocs/rtd_tests/tests/test_doc_building.py:1195]: Are we checking in another place of the test that we are installing `setuptools==41.0.1`? I think there should be an update for a test case, right?

File: readthedocs/doc_builder/python_environments.py [hunk 2]
```diff
@@ -337,21 +337,25 @@ def install_core_requirements(self):
...
-        cmd = pip_install_cmd + [pip_version]
+        setuptools_version = self.project.get_feature_value(
+            Feature.INSTALL_LATEST_SETUPTOOLS,
+            positive='setuptools',
+            negative='setuptools==41.0.1',
+        )
+        cmd = pip_install_cmd + [pip_version, setuptools_version]
         self.build_env.run(
             *cmd, bin_path=self.venv_bin(), cwd=self.checkout_path
         )
```
Comment (humitos) [inline readthedocs/doc_builder/python_environments.py:356]: We are changing _where_ we install `setuptools` here.

Now we are installing it together with `pip` as the first command we executed. Before, we were installing it together with our own dependencies. I don't think this is something bad, but definitely a change that may break in a random way :upside_down_face: 

It makes sense to be installed with `pip` to me; but we should keep an eye on this.

**Review by humitos — APPROVED** [review]
Looks good to me to use a feature flag for this.

**Pull request merged**